ChaosMod/Effects/db/Player/PlayerPacifist.cpp

```diff
@@ -2,19 +2,43 @@
 //Effect by ProfessorBiddle, requested thousands of times on discord
 static void OnTick()
 {
-	Player player = PLAYER_ID();
+	int hits0, hits1, hits2;
+	STAT_GET_INT(GET_HASH_KEY("SP0_HITS"), &hits0, -1);
+	STAT_GET_INT(GET_HASH_KEY("SP1_HITS"), &hits1, -1);
+	STAT_GET_INT(GET_HASH_KEY("SP2_HITS"), &hits2, -1);
```

Comment:
wouldnt it make more sense to make it so:
* you hold the current player hash and current hits 
* Check on Tick if the player changed
  * if so, set the current hits to the current player hits 
* if not, compare the current hits to the last saved hits
  * if > last hits -> kill
  * if <= hits -> set to saved hits (may happen after mission restart / loading a savegame?)
* Also, ONLY get the hash within the switch case, the rest is always the same procedure and just duplicates the code

Just my 2 cents

Reply:
Thanks for the input, I'll do this.